base/file.jl

```diff
@@ -356,13 +339,42 @@ is an open file object for this path.
 mktemp(parent)
 
 """
-    mktempdir(parent=tempdir())
+    mktempdir(parent=tempdir(); prefix="$(repr(temp_prefix))")
 
 Create a temporary directory in the `parent` directory and return its path.
 If `parent` does not exist, throw an error.
+An optional `prefix` to the directory name can be provided.
 """
-mktempdir(parent)
+function mktempdir(parent=tempdir(); prefix=temp_prefix)
+    i = endof(parent)
+    while i >= 1 && parent[i:i] == path_separator
+        i -= 1
+    end
+    parent = parent[1:i]
+    i = 1
+    while i <= endof(prefix) && prefix[i:i] == path_separator
+        i += 1
+    end
+    prefix = prefix[i:end]
 
+    tpath = "$(parent)$(path_separator)$(prefix)XXXXXX"
```

Comment:
If `prefix` contains the letter `X` this is going to be wrong; it's hard to say if that's avoidable or not, given the way the underlying `uv_fs_mkdtemp` API works.

Reply:
Yes. It's not even clear why these chars need to be `X` rather than any other ASCII char.

Reply:
Links to documentation for `uv_fs_mkdtemp` and `mkdtemp` given below:

http://docs.libuv.org/en/v1.x/fs.html
https://linux.die.net/man/3/mkdtemp